travis/travis_weblate.py

```diff
@@ -1,166 +1,242 @@
+                   "row['translation'] = ''/g\" %s" % s_file)
+            print cmd
+            subprocess.call(cmd, shell=True)
+
+    def _get_modules_installed(self):
+        self._installed_modules = []
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            odoo_context.cr.execute("select name from ir_module_module"
+                                    " where state = 'installed'")
+            self._installed_modules = odoo_context.cr.dictfetchall()
+
+    def _generate_odoo_po_files(self, component):
+        generated = False
+        module = []
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            for install in self._installed_modules:
+                if install['name'] in component['filemask']:
+                    module = install['name']
+            if not module:
+                return generated
             print("\n", yellow("Obtaining POT file for %s" % module))
-            i18n_folder = os.path.join(travis_build_dir, module, 'i18n')
-            # # TODO: Add empty es.po files if non exists
-            # source_filename = os.path.join(i18n_folder, module + ".pot")
-            # # Create i18n/ directory if doesn't exist
+            i18n_folder = os.path.join(self._travis_build_dir, module, 'i18n')
             if not os.path.isdir(i18n_folder):
                 os.makedirs(i18n_folder)
-            # with open(source_filename, 'w') as f:
-            #     f.write(odoo_context.get_pot_contents(module))
-
             # Put git add for letting known git which translations to update
-            for po_file_name in os.listdir(i18n_folder):
-                if not po_file_name.endswith('.po'):
+            po_files = glob.glob(os.path.join(i18n_folder, '*.po'))
+            for lang in self._langs:
+                if os.path.isfile(os.path.join(i18n_folder, lang + '.po')):
                     continue
-                lang = os.path.splitext(po_file_name)[0]
-                if langs and lang not in langs:
+                with open(os.path.join(i18n_folder, lang + '.po'), 'wb')\
+                        as f_po:
+                    f_po.write(odoo_context.get_pot_contents(module, lang))
+            for po_file_name in po_files:
+                lang = os.path.basename(os.path.splitext(po_file_name)[0])
+                if self._langs and lang not in self._langs:
                     # Limit just allowed languages if is defined
                     continue
                 po_file_path = os.path.join(i18n_folder, po_file_name)
                 with open(po_file_path, 'r') as f_po:
-                    current_content = f_po.read()
                     odoo_context.load_po(f_po, lang)
                 new_content = odoo_context.get_pot_contents(module, lang)
```

Comment:
The method `load_po` load the translations and the method `get_pot_contents` get the content